Utils/PluginActivator/source/COMRPCStarter.cpp

```diff
@@ -0,0 +1,149 @@
+        // Normally this is because either:
+        //  a) Thunder isn't running or we can't connect to /tmp/communicator
+        //  b) A plugin with the specified callsign does not exist
+        if (!shell) {
+            LOG_ERROR(mPluginName.c_str(), "Failed to open IShell interface for %s", mPluginName.c_str());
+            currentRetry++;
+
+            // Must close the connection before we retry, since even if shell is null the underlying
+            // connection will still have been opened
+            mClient->Close(RPC::CommunicationTimeOut);
+
+            std::this_thread::sleep_for(std::chrono::milliseconds(retryDelayMs));
+            continue;
+        };
+
+        // Connected to Thunder successfully and got the plugin shell
+        if (shell->State() == PluginHost::IShell::ACTIVATED) {
+            LOG_INF(mPluginName.c_str(), "Plugin is already activated - nothing to do");
+            success = true;
+        } else {
+            // Will block until plugin is activated
+            uint32_t result = shell->Activate(PluginHost::IShell::REQUESTED);
+
+            auto end = std::chrono::steady_clock::now();
+            auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
+
+            if (result != Core::ERROR_NONE) {
+                if (result == Core::ERROR_PENDING_CONDITIONS) {
+                    // Ideally we'd print out which preconditions are un-met for debugging, but that data is not exposed through the IShell interface
+                    LOG_ERROR(mPluginName.c_str(), "Failed to activate plugin due to unmet preconditions after %ldms", duration.count());
+                } else {
+                    LOG_ERROR(mPluginName.c_str(), "Failed to activate plugin with error %u (%s) after %ldms", result, Core::ErrorToString(result), duration.count());
+                }
+                // Try activation again up until the max number of retries
+                currentRetry++;
+                std::this_thread::sleep_for(std::chrono::milliseconds(retryDelayMs));
+                continue;
+            } else {
+                // Our work here is done!
+                LOG_INF(mPluginName.c_str(), "Successfully activated plugin after %ldms", duration.count());
+                success = true;
+            }
+        }
+    }
+
+    if (!success) {
+        LOG_ERROR(mPluginName.c_str(), "Max retries hit - giving up activating the plugin");
+    }
+
+    if (shell) {
+        shell->Release();
+    }
+
+    return success;
+}
+
+/**
+ * Retrieves the socket we will communicate over for COM-RPC
+ */
+Core::NodeId COMRPCStarter::getConnectionEndpoint() const
+{
+    string communicatorPath;
+    Core::SystemInfo::GetEnvironment(_T("COMMUNICATOR_PATH"), communicatorPath);
+
+    // On linux, Thunder defaults to /tmp/communicator for the generic COM-RPC
+    // interface
+    if (communicatorPath.empty()) {
+        communicatorPath = "/tmp/communicator";
```

Comment:
Must admit, current code base is also not that strict on it (and we have never tried to actually run/build wchar_t cde generation, but the missing _T("") macro would definitly cause a performance penalty or a warning/error here..